src/components/views/rooms/MessageComposerInput.js

```diff
@@ -36,49 +38,30 @@ var sdk = require('../../../index');
 var dis = require("../../../dispatcher");
 var KeyCode = require("../../../KeyCode");
 
+import {contentStateToHTML, HTMLtoContentState, getScopedDecorator} from '../../../RichText';
+
 var TYPING_USER_TIMEOUT = 10000;
 var TYPING_SERVER_TIMEOUT = 30000;
 var MARKDOWN_ENABLED = true;
 
-function mdownToHtml(mdown) {
-    var html = marked(mdown) || "";
-    html = html.trim();
-    // strip start and end <p> tags else you get 'orrible spacing
-    if (html.indexOf("<p>") === 0) {
-        html = html.substring("<p>".length);
-    }
-    if (html.lastIndexOf("</p>") === (html.length - "</p>".length)) {
-        html = html.substring(0, html.length - "</p>".length);
-    }
-    return html;
-}
-
 /*
  * The textInput part of the MessageComposer
  */
-module.exports = React.createClass({
-    displayName: 'MessageComposerInput',
-
-    statics: {
-        // the height we limit the composer to
-        MAX_HEIGHT: 100,
-    },
-
-    propTypes: {
-        tabComplete: React.PropTypes.any,
-
-        // a callback which is called when the height of the composer is
-        // changed due to a change in content.
-        onResize: React.PropTypes.func,
-
-        // js-sdk Room object
-        room: React.PropTypes.object.isRequired,
-    },
+export default class MessageComposerInput extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.onAction = this.onAction.bind(this);
+        this.onInputClick = this.onInputClick.bind(this);
+
+        this.state = {
+            editorState: EditorState.createEmpty(getScopedDecorator(this.props))
+        };
+    }
 
-    componentWillMount: function() {
+    componentWillMount() {
         this.oldScrollHeight = 0;
         this.markdownEnabled = MARKDOWN_ENABLED;
-        var self = this;
+        const component = this;
```

Comment:
I don't like how we're breaking the de-facto alias used in Vector for `this` (`self`) for `component`, for seemingly no reason. Is there any reason for doing this? If not, either remove this entirely and use arrow functions, or use `self` please.